contracts/flow/token/TeleportedTetherToken.cdc

```diff
@@ -104,101 +154,158 @@ pub contract TeleportedTetherToken: FungibleToken {
   // and store the returned Vault in their storage in order to allow their
   // account to be able to receive deposits of this token type.
   //
-  pub fun createEmptyVault(): @FungibleToken.Vault {
+  access(all) fun createEmptyVault(vaultType: Type): @TeleportedTetherToken.Vault {
     return <- create Vault(balance: 0.0)
   }
 
-  pub resource Allowance {
-    pub var balance: UFix64
+  // Gets a list of the metadata views that this contract supports
+  access(all) view fun getContractViews(resourceType: Type?): [Type] {
+    return [Type<FungibleTokenMetadataViews.FTView>(),
+            Type<FungibleTokenMetadataViews.FTDisplay>(),
+            Type<FungibleTokenMetadataViews.FTVaultData>(),
+            Type<FungibleTokenMetadataViews.TotalSupply>()]
+  }
+
+  // Get a Metadata View from TeleportedTetherToken
+  //
+  // @param view: The Type of the desired view.
+  // @return A structure representing the requested view.
+  //
+  access(all) fun resolveContractView(resourceType: Type?, viewType: Type): AnyStruct? {
+    switch viewType {
+      case Type<FungibleTokenMetadataViews.FTView>():
+        return FungibleTokenMetadataViews.FTView(
+          ftDisplay: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
+          ftVaultData: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+        )
+      case Type<FungibleTokenMetadataViews.FTDisplay>():
+        let media = MetadataViews.Media(
+          file: MetadataViews.HTTPFile(
+            url: "https://raw.githubusercontent.com/blocto/assets/main/color/flow/tusdt.svg"
+          ),
+          mediaType: "image/svg+xml"
+        )
+        let medias = MetadataViews.Medias([media])
+        return FungibleTokenMetadataViews.FTDisplay(
+          name: "teleported USDT",
+          symbol: "tUSDT",
+          description: "tUSDT stands for teleported USDT on Flow Blockchain.",
+          externalURL: MetadataViews.ExternalURL("https://blocto.io"),
+          logos: medias,
+          socials: {
+            "twitter": MetadataViews.ExternalURL("https://twitter.com/BloctoApp")
+          }
+        )
+      case Type<FungibleTokenMetadataViews.FTVaultData>():
+        let vaultRef = TeleportedTetherToken.account.storage.borrow<auth(FungibleToken.Withdraw) &TeleportedTetherToken.Vault>(from: /storage/TeleportedTetherTokenVault)
```

Comment:
`(from: /storage/teleportedTetherTokenVault)`

Reply:
typo

Reply:
Also in the contract init method, you did not create the vault
```
self.account.storage.save(<-vault, to: /storage/teleportedTetherTokenVault)
```
Here you will run into 
```
panic("Could not borrow reference to the contract's Vault!")
```

Reply:
Is it necessary to create the vault in the `init` method? As I know, the contract owner (administrator) might not need the vault.
If it’s needed, a transaction can be sent later to create the vault.

Reply:
Got your point, will fix it.

Reply:
I sent that transaction on testnet to initiate the tUSDT token vault on this contract BTW :P
So I can work on the tUSDT swap functionalities.

Reply:
fixed.